caliper_sender.py

```diff
@@ -134,39 +142,53 @@ def send_caliper_event():
 
 
 def get_caliper_event(event, event_type, event_action):
-    nav_path = document_path = chapter_path = ""
-    rsc = {}
-    if event.get('div_id'):
-        nav_path = event.get('div_id').split('/')
-        document_path = '/'.join(nav_path[:4]) + '/'
-        chapter_path = '/'.join(nav_path[:5]) + '/'
-        if len(nav_path) > 3:
-            rsc['document'] = nav_path[3]
-        if len(nav_path) > 4:
-            rsc['chapter'] = nav_path[4]
-        if len(nav_path) > 5:
-            rsc['page'] = nav_path[5]
+    # It's possible to not have chapters if it's a top level. This would result in -3 being 'published'
+    if not event.get('div_id'):
+        return None
+    nav_path = event.get('div_id').split('/')
+    if len(nav_path) < 3:
+        return None
+
+    # Need to try to figure out the document/chapter/page from the div id
+    # If there's no scheme, add the file: scheme
+    if not nav_path[0]:
+        nav_path[0] = 'file:'
+    document_path = '/'.join(nav_path[:-2]) + '/'
+    chapter_path = '/'.join(nav_path[:-1]) + '/'
+
+    # On the index page there may not be a chapter, just the document. The only way I can seem to tell this is if the name is published
+    if 'published' == nav_path[-3]:
+        is_part_of = caliper.entities.Document(
+            id = document_path,
+            name = nav_path[-2],
+        )
+    # Otherwise the chapter is part of the document
+    else:
+        is_part_of = caliper.entities.Chapter(
+            id = chapter_path,
+            name = nav_path[-2],
+            isPartOf = caliper.entities.Document(
+                id = document_path,
+                name = nav_path[-3],
+            )
+        )
 
+    # Page is either part of chapter & document or just part of the document
     resource = caliper.entities.Page(
         id='/'.join(nav_path),
-        name=rsc.get('page'),
-        isPartOf=caliper.entities.Chapter(
-            id=chapter_path,
-            name=rsc.get('chapter'),
-            isPartOf=caliper.entities.Document(
-                id=document_path,
-                name=rsc.get('document'),
-            )
-        )
+        # Page name is always the last item
+        name= nav_path[-1],
     )
 
-    actor = caliper.entities.Person(id=event.get('sid'))
+    resource.isPartOf = is_part_of
+
+    actor = caliper.entities.Person(id="urn:actor_id:" + event.get('sid'))
     course_id = os.getenv("COURSE_ID")
     edapp_id = os.getenv("EDAPP_ID")
     if not course_id or not edapp_id:
         raise Exception("You need to define both EDAPP_ID and COURSE_ID before using this.")
-    organization = caliper.entities.Organization(id=course_id)
-    edApp = caliper.entities.SoftwareApplication(id=edapp_id)
+    organization = caliper.entities.Organization(id="urn:course_offering_id" + course_id)
```

Comment:
a ":" is missed at the end of ""urn:course_offering_id"

Reply:
```suggestion
    organization = caliper.entities.Organization(id="urn:course_offering_id:" + course_id)
```

Reply:
Thanks!